### ISSUE TITLE
[cct3 plugin] crash will different molecules in same input, e.g. in NBody calculations

### ISSUE THREAD

**Issue by hokru** [psi4/psi4#2009]
This might be rather an issue for https://github.com/piecuch-group/cct3 but will affect psi4 users nonetheless.

The following is a minimal input:
```
import cct3

set {
basis cc-pVDZ
ints_tolerance 1e-11
}

molecule eins {
units bohr
symmetry c1
h -1.000000000000  -2.414213562373   0.000000000000
h -1.000000000000   2.414213562373   0.000000000000
h  1.000000000000  -2.414213562373   0.000000000000
h  1.000000000000   2.414213562373   0.000000000000
}

molecule zwei {
units bohr
symmetry c1
h -2.514213562373  -1.000000000000   0.000000000000
h -2.514213562373   1.000000000000   0.000000000000
h  2.514213562373  -1.000000000000   0.000000000000
h  2.514213562373   1.000000000000   0.000000000000
h -1.000000000000  -2.414213562373   0.000000000000
h -1.000000000000   2.414213562373   0.000000000000
h  1.000000000000  -2.414213562373   0.000000000000
h  1.000000000000   2.414213562373   0.000000000000
}


energy("cr-cc(2,3)",molecule=eins)
clean()
energy("cr-cc(2,3)",momlecule=zwei)
```

The error is:
```
(base) kruse@sphinx1 /scratch/kruse/CCT3 $ psi4.git  -v -s . -n 16 input.dat
forrtl: severe (104): incorrect RECL= specifier value for connected file, unit 250, file /scratch/kruse/CCT3/PA
Image              PC                Routine            Line        Source
libifcoremt.so.5   00002AD86592CE75  for__io_return        Unknown  Unknown
libifcoremt.so.5   00002AD8659522E0  for_open              Unknown  Unknown
cct3.so            00002AD865620EDF  integrals_mp_load     Unknown  Unknown
cct3.so            00002AD86562C3EB  solver_mp_solve_c     Unknown  Unknown
cct3.so            00002AD8655F75A6  driver_mp_run_cc_     Unknown  Unknown
cct3.so            00002AD86563683E  cc_                   Unknown  Unknown
cct3.so            00002AD85AD0AAE1  cct3                  Unknown  Unknown
```

The `PA` file is opened for _writing_ with a record length `recl`  using orbital dimensions.

**Comment by andysim** [comment]
Just a quick check - is the `momlecule` typo on the last line just in this PR, or in the input file?  I don't think that's the cause, but wanted to make sure

**Comment by hokru** [comment]
Ups, a proper typo in the original input, but the right molecule was used and the error persists.

----

I think I identified the (or a) problem. There are direct access fortran files that are never closed as far as I tracked it down. If I do so, it seems to work.

**Comment by andysim** [comment]
Aaah, great work!  If it's fortran direct access, rather than DPD, that should hopefully be easy to cure with some `close` statements in the plugin 🤞

**Comment by hokru** [comment]
PR submitted https://github.com/piecuch-group/cct3/pull/5

**Comment by hokru** [comment]
PR with my fix is merged. Leaving this open until our cmake pulls the new plugin version @loriab

**Comment by hokru** [comment]
resolved by #2065
